core/Tracker/GoalManager.php

```diff
@@ -99,7 +99,7 @@ public static function getGoalDefinition($idSite, $idGoal)
         $goals = self::getGoalDefinitions($idSite);
 
         foreach ($goals as $goal) {
-            if ($goal['idgoal'] == $idGoal) {
+            if ($goal['idgoal'] === $idGoal) {
```

Comment:
$goals is coming from database, so the values might be integer or string. Same applies for the value given to that method:
```suggestion
            if ((int) $goal['idgoal'] === (int) $idGoal) {
```